File: binary-search.rs
pub fn find<T : Ord , U : AsRef<[T]>>(array: U, key: T) -> Option<usize> {
    let array = array.as_ref();
    let mut left: i32 = 0;
    let mut right = (array.len() as i32 - 1) as i32;
    while right >= left {
        let mid = left + (right - left)/2;
        if array[mid as usize] == key {return Some(mid as usize);}
        if array[mid as usize] > key {right = mid - 1;}
        else {left = mid + 1;}
    }
    None
}

File: solutions/python/binary-search/1/binary_search.py
def find(search_list, value):
    left = 0
    right = len(search_list) - 1
    while right >= left:
        mid = (left + right) // 2
        if value <= search_list[mid]:
            right = mid - 1
        else:
            left = mid + 1
    if left < len(search_list) and search_list[left] == value:
        return left
    raise ValueError("value not in array")
        
        
